src/os/data/registry.js

```diff
@@ -0,0 +1,86 @@
+goog.module('os.data.Registry');
+
+
+/**
+ * Object wrapper to serve as agnostic manager of objects
+ * @template T
+ */
+class Registry {
```

Comment:
This class is similar to https://github.com/ngageoint/opensphere/blob/master/src/os/data/collectionmanager.js. You may even be able to replace it entirely and simply use the existing one.

Reply:
Ended up needing some of the extended abilities in Registry; but added functions and made it an EventTarget just like CollectionManager.